### DISCUSSION TITLE
Further improvement to entry workflow UX

### DISCUSSION THREAD

**Discussion by FreekVR** [craftcms/cms#8994] [Ideas]
### Description

The recent changes to entry workflow are really nice, and I think the change to show Drafts in the entries index is a nice addition.

Based on internal and feedback from our customers I'd like to suggest some additional changes to clarify the various states an entry has.

The problems we hear about;
- What is meant by the entry status (live/pending/expired/disabled) is unclear in the entries index list
- The various revisions for entries and the fact that one can be published but still have an unpublished draft can be a bit confusing (esp now that the default action is "save draft")
- It isn't always clear for which site you are viewing/editing entries
- It isn't always clear whether bulk changes made to entries from the entries listing apply to only the selected site, or all sites
- People miss the fact that there's filtering and sorting options 

I think there's actually a few purely visual changes which could help clarify:
- Make the dropdowns for the selected site stand out more, in all places, or display the currently selected site more distinctively in some other way
- Add tooltips and/or visually legible labels to the site/filter/sorting dropdowns on top of the entries list (so users can see what they do, rather than just see the default value and attempt to reason from there)
- Add tooltips and/or a secondary label (eg. a very short help-text) to the values in the filter dropdown AND to the the status bullet in the listing
- Display filters, search. etc on top of the right column showing the entries (since that is what they apply to) instead of as a wide header that appears apart from the entries list table

Perhaps you have some other things in mind here, but I just thought I'd start this discussion to see how we may improve.

Thanks for reading along thus far  :)

**Comment by brandonkelly** [comment]
> What is meant by the entry status (live/pending/expired/disabled) is unclear in the entries index list

How is it unclear? The statuses seem pretty distinct to me:

- **Live**: The entry is enabled, and is currently live on the website
- **Pending**: The entry is enabled, and will be live once its Publish Date has been reached
- **Expired**: The entry is enabled, but has an Expiry Date which has been reached
- **Disabled**: The entry is disabled

> The various revisions for entries and the fact that one can be published but still have an unpublished draft can be a bit confusing (esp now that the default action is "save draft")

What do you mean by “the fact that one can be published but still have an unpublished draft”?

Entries primarily exist in one of two states:

- **Unpublished draft**: the entry has never been published before.
- **Normal**: the entry has been published.

Perhaps the confusion comes from the word “published” – which doesn’t necessarily mean the entry is _live_. You can “publish” an entry with a disabled status, and it will no longer be in the “unpublished draft” state, however it is not _live_ either. We’ve spent a lot of time trying to think of a better word than “publish” (e.g. “commit”), but nothing has felt like an obvious improvement. If you have a better idea i’m all ears.

**Comment by FreekVR** [comment]
> How is it unclear? The statuses seem pretty distinct to me:
> Live: The entry is enabled, and is currently live on the website
> Pending: The entry is enabled, and will be live once its Publish Date has been reached
> Expired: The entry is enabled, but has an Expiry Date which has been reached
> Disabled: The entry is disabled

Think of it from an inexperienced users' perspective, where they may not be aware that these features exist (yet) and they are not explained at all on the entry listing. I think even adding those descriptions you just posted would be enough to clarify, and would perhaps work well as a tooltip for the bullets or as a legible (sub-)label. Also people tend to "miss" the dropdown with the status filters, and that's the only place where the bullet colours actually get explained AFAIK.

> What do you mean by “the fact that one can be published but still have an unpublished draft”?

I think people get slightly confused with the green bullet showing up for new entries posted as drafts, which they interpret as "published on the website" - which it is not. I can imagine the same would go for "pending" which would indicate a draft would get automatically published once its publication date is reached,, but I believe that is also not the case (unless said draft is published first)

For existing entries, sometimes I get the feedback that they think "save as draft" turns an existing published entry into a draft, which would unpublish it, while in fact it of course only adds a *new* unpublished draft revision for an entry, with the original still being published/live.

I think it would perhaps be helpful to add additional statuses in the overview somehow, eg. for "**Unpublished draft**: No drafts for this entry have been published yet" and "**Has draft(s):** This entry is [live/pending/expired/disabled] but has additional changes in drafts which are not published"

**Comment by brandonkelly** [comment]
> I think people get slightly confused with the green bullet showing up for new entries posted as drafts

Ah, I see what you mean. We ended up changing it so drafts don’t get a colored status icon in today’s Craft 3.6.4 release, so that should help.

> For existing entries, sometimes I get the feedback that they think "save as draft" turns an existing published entry into a draft, which would unpublish it, while in fact it of course only adds a _new_ unpublished draft revision for an entry, with the original still being published/live.

Interesting, thanks.

**Comment by JeanLucEsser** [comment]
Just saw this issue after I wrote mine. 
Referencing it here (#7517) as I think they could relate in a way.

**Comment by JeanLucEsser** [comment]
I'll add my comments right here though.

> What is meant by the entry status (live/pending/expired/disabled) is unclear in the entries index list

That was my issue as well when I started thinking about how to improve all this.
There are actually two type of statuses that somehow got conflated with 3.6.3:

- Entry statuses : current (published) / drafts / revisions / trashed
- Publish statuses : live / pending / expired / disabled

Before 3.6.3, it was clear that publish status was only for current (published) entries.
But now, by adding drafts into the mix, it's not that clear anymore (in the entries list I mean).

3.6.4 decided to bypass the publish status for drafts. It's less confusing but still not perfect.

Replacing the publish status dot with the draft icon would be better because it would say "we are not showing you the publish status because it doesn't makes sense yet as this is still a work in progress", instead of saying "we are showing you a disabled publish status (or is it meant as all?) even if you prepared your draft for live because, well, because".

And it would allow for removing of the DRAFT tag which I'm not a fan of (replace it with the draft name in a gray text).

> The various revisions for entries and the fact that one can be published but still have an unpublished draft can be a bit confusing (esp now that the default action is "save draft")

I agree here, sometimes you are working on multiple drafts and when publishing, you'd like to get rid of all your trials and errors.

I'm in the process of writing an issue for that, offering an option to 'Publish Draft' in the way of 'Publish and remove all pending drafts'.

> It isn't always clear for which site you are viewing/editing entries
> It isn't always clear whether bulk changes made to entries from the entries listing apply to only the selected site, or all sites

I do agree in a way, but it's not a big issue per se.
I have some ideas for that, which I didn't fully laid out yet.

One of them would be to use the translate icon near the title field to open a compare and merge tool in which you could select a version of the field content from the published entry, a draft, a revision, and even a different site, and compare and merge it into the actual field. But that would be a HUGE undertaking.

> People miss the fact that there's filtering and sorting options

I had these kind of feedback but once I show them, they're good to go.
Not sure what could be improved here... rethinking the search bar while keeping it as a one liner is not a simple task.

I do have ideas on this, including saved and persistent searches and filters, different column arrangement depending on the sorting applied (thing of structured content which is displayed differently when sorting is structured). It's in a corner of my mind, and I'll lay it all out one day, I promise.

Thanx for reading (English is not my mother tongue so BIG thanx actually).

**Comment by FreekVR** [comment]
I think re. the site status and filter dropdowns both adding a (placeholder/tooltip)-label AND tweaking its background color to have more contrast and feel more like a dropdown would help a lot actually and that shouldn't be too hard to add, I think it just fades into the background a bit too much for people to notice straight away (show, don't tell ;))

(Generally speaking, in UX/A11Y world I don't think its generally a good idea to have unlabeled form controls, which I consider these to be, but I understand that has some layout implication in the current setup)
